server/chat/background/task.py

```
@@ -513,103 +514,109 @@ def run_background_chat(
+                                (incident_id,)
                             )
-                        else:
-                            logger.info(
-                                f"[BackgroundChat] Linked session {session_id} and task {self.request.id} to incident {incident_id}"
+                            row = cursor.fetchone()
+                            existing_task_id = row[0] if row and row[0] else None
+
+                            if existing_task_id and existing_task_id != self.request.id:
+                                logger.warning(
+                                    f"[BackgroundChat] Incident {incident_id} already has task ID {existing_task_id}, "
+                                    f"but this task is {self.request.id}. This may indicate a race condition or duplicate RCA start."
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)